### DISCUSSION TITLE
Using string functions to define virtual columns

### DISCUSSION THREAD

**Discussion by elcapo** [tpetry/laravel-query-expressions#9] [Q&A] [answered]
Hi @tpetry,

I'm trying to use some of the string functions to define virtual columns and, if I try the way I'd expect to use the expressions:

~~~php
$table->string('full_name')
    ->virtualAs(
        (new Concat([
            new Coalesce(['first_name', '']),
            new Coalesce(['last_name', '']),
        ]))
    );
~~~

I get an ugly error message:

~~~
Object of class Tpetry\QueryExpressions\Function\String\Concat could not be converted to string
~~~

To fix it, I need to explicitly call **getValue**, as in:

~~~php
$grammar = DB::connection()->getQueryGrammar();

$table->string('full_name')
    ->virtualAs(
        (new Concat([
            new Coalesce(['first_name', '']),
            new Coalesce(['last_name', '']),
        ]))->getValue($grammar)
    );
~~~

Is this an expected behavior? Maybe we could define a helper to use the expressions in this context without having to manually pass them the grammar?

**Comment by tpetry** [comment, marked as answer]
That is not expected. It is a bug in Laravel. I will work on fix network 😉
